inc/tcpServer.hpp

```diff
@@ -59,6 +59,23 @@ class TcpServer {
 		}
 	};
 
+	void send(std::vector<char> data, size_t nbytes) {
+		std::vector<char> sendBuffer(data);
+		sendBuffer.resize(nbytes);
+		socket.async_send(boost::asio::buffer(sendBuffer, nbytes), 
+			[](const boost::system::error_code& error, std::size_t bytes_transferred) {
+				if (!error) {
+					std::cerr << "TCP socket: " << bytes_transferred << " bytes sent, " << std::endl;
+					throw boost::system::system_error(boost::asio::error::eof);
```

Comment:
Why do you want to throw an exception on success?

Reply:
Don't know how that got in there. I changed it in a new commit